### PR TITLE
Check for translation updates

### PR DESCRIPTION
**Pull request by di**
This adds a CI job to check whether the translations need to be updated, as well as a more user-friendly command, `make translations`.

It also recompiles the POT file to get the job passing.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -177,9 +177,7 @@ stop:
 compile-pot: .state/env/pyvenv.cfg
 	PYTHONPATH=$(PWD) $(BINDIR)/pybabel extract \
 		-F babel.cfg \
-		--copyright-holder="PyPA" \
-		--msgid-bugs-address="https://github.com/pypa/warehouse/issues/new" \
-		--project="Warehouse" \
+		--omit-header \
```
Comment (ewdurbin) [inline Makefile]: good call here.

**Comment by ewdurbin** [comment]
@di I think omitting the location notes might break the "Nearby strings" feature in weblate?
Screen Shot 2020-02-25 at 3 42 09 PM

**Comment by di** [comment]
@ewdurbin I think it might. Tradeoffs... Those strings don't look super helpful for that particular translation (but I have not actually used Weblate so I can't really say).

File: Makefile [hunk 2]
```diff
@@ -177,9 +177,8 @@ stop:
...
+		--no-location \
```
Comment (webknjaz) [inline Makefile]: I think it'd be a good idea to maybe have separate targets just for identifying if regeneration of `messages.pot` is needed and keep the original one as is.

Comment (webknjaz) [reply Makefile]: Location info is actually useful as it allows us to track down places in the templates if the context is unclear.

File: .travis.yml [hunk 3]
```diff
@@ -55,3 +55,6 @@ matrix:
     - script: make licenses BINDIR="$(dirname $(which python))"
       env:
         - SUITE=Licenses
+    - script: make translations BINDIR="$(dirname $(which python))"
```
Comment (webknjaz) [inline .travis.yml]: The alternative could look like this: https://github.com/pypa/warehouse/issues/7409#issuecomment-591443217. Having an up-to-date translations source is more useful in master after the PR merge, not in PRs themselves.

**Comment by di** [comment]
I added the location comments back, we'll see how annoying it is.

**Comment by webknjaz** [comment]
I confirm that new strings appeared in the UI and I got a notification from Weblate about it.

**Pull request merged**